wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -633,9 +633,11 @@ public final class ZMUserSession: NSObject {
 
     // temporary function to simplify call to ConversationEventProcessor
     // might be replaced by something more elegant
-    public func processConversationEvents(_ events: [ZMUpdateEvent]) {
-        WaitingGroupTask(context: self.syncContext) {
-            await self.conversationEventProcessor.processConversationEvents(events)
+    public func processConversationEvents(_ events: [ZMUpdateEvent], completion: (() -> Void)?) {
+        WaitingGroupTask(context: self.syncContext) { [weak self] in
+            guard let self else { return }
+            await self.conversationEventProcessor.processAndSaveConversationEvents(events)
+            completion?()
```

Comment:
Shouldn't `completion` be called always?
```suggestion
        WaitingGroupTask(context: self.syncContext) { [weak self] in
            defer { completion?() }
            guard let self else { return }
            await self.conversationEventProcessor.processAndSaveConversationEvents(events)
            completion?()
```
or
```suggestion
        WaitingGroupTask(context: self.syncContext) { [weak self] in
            guard let self else { return completion?() }
            await self.conversationEventProcessor.processAndSaveConversationEvents(events)
            completion?()
```